docs/structured-streaming-kafka-integration.md

```diff
@@ -310,6 +310,23 @@ The following configurations are optional:
 
 <table class="table">
 <tr><th>Option</th><th>value</th><th>default</th><th>query type</th><th>meaning</th></tr>
+<tr>
```

Comment:
1. What timestamp field contains differs significantly depending on Kafka's `message.timestamp.type` configuration. The situation is even worse when a topic migrated into a new topic (because of transformation or something).
2. As I've seen users are dealing with problems which are making filtering based on event time but this field is most of the time processing time. This has to be mentioned.

Some compacted limitations would be good add.

Reply:
For 1), given that this is a Spark document I would say it's better to not explain Kafka stuff deeply. Kafka should have specific documentation regarding this (even KIP) and end users should rely on that to understand how it works. 

We could refer the doc if we find any (I'll try to find one hopefully not KIP), but even there's no doc, we could mention this feature leverage `KafkaConsumer.offsetsForTimes ` and refer to javadoc on Kafka API.

For 2), I agree it's ideal to mention timestamp offset here is related to when message is ingested, but I also feel it may be a bad signal that they are not understanding either Kafka's `offsetsForTimes` or Spark's processing time vs event time if end users don't know this before mentioned it. Given both have docs for each so might be redundant if we mention this here again. But that's just my 2 cents and if we can explain it as one-liner or two it may not be too redundant.

Reply:
I agree too detailed explanation is bad/redundant and that's the reason why I've suggested `compacted limitations`.

Stating `The start point of timestamp when a query is started` is just not true because producer can overwrite this field under some circumstances (not Spark is the only producer). I would refer to Kafka configuration and mention that this field is depending on how Kafka is configured / data produced without mentioning all the details. I think based on the field users can look for KIP in need.

Yeah, offsetsForTimes is not over-documented but that's a Kafka question.

Reply:
This can be marked as resolved, can't be?

Reply:
I feel you're the one to mark conversation as resolved, since you've initiated the conversation and you're the one to evaluate the conversation. Please let me know if you can't make it because of Github UI.

Reply:
No resolve button :)

Reply:
Ah! I recall you've commented and marked the conversations which you initiated as resolved in other PR. Wondering why it doesn't work for this time. Thanks for clarifying! Will mark as resolved.